com.unity.ml-agents.extensions/Runtime/Sensors/ArticulationBodySensorComponent.cs

```diff
@@ -34,12 +34,10 @@ public override int[] GetObservationShape()
             var poseExtractor = new ArticulationBodyPoseExtractor(RootBody);
             var numPoseObservations = poseExtractor.GetNumPoseObservations(Settings);
             var numJointObservations = 0;
-            // Start from i=1 to ignore the root
-            for (var i = 1; i < poseExtractor.Bodies.Length; i++)
```

Comment:
This wouldn't account for non-root bodies being disabled (or the root being enabled)